datafusion/core/src/datasource/physical_plan/csv.rs

```diff
@@ -112,6 +114,7 @@ impl CsvExecBuilder {
             has_header: false,
             delimiter: b',',
             quote: b'"',
+            terminator: None,
```

Comment:
I wonder if this option should have any interactions with `newlines_in_values`, or somehow affect execution of CSVExec  (if `newlines_in_values` used anywhere to actually check input data -- I wasn't able to find such a usages for it, but still not sure).

Reply:
I did some tests. I think that `newlines_in_values` won't be impacted by `terminator`.  No matter how I change the terminator, the number of rows is the same.

```rust
        let session_ctx = SessionContext::new();
        let store = object_store::memory::InMemory::new();

        let data = bytes::Bytes::from("a,b\r1,\"2\na\"\r3,\"4\nb\"");
        let path = object_store::path::Path::from("a.csv");
        store.put(&path, data.into()).await.unwrap();

        let url = Url::parse("memory://").unwrap();
        session_ctx.register_object_store(&url, Arc::new(store));

        let df = session_ctx
            .read_csv("memory:///", CsvReadOptions::new().newlines_in_values(true).terminator(Some(b'\r')))
            .await
            .unwrap();

        let result = df.collect().await.unwrap();
        assert_eq!(result[0].num_rows(), 2);
```